ludwig/datasets/README.md

```diff
@@ -14,24 +15,82 @@ dataset_df = titanic.load()
 train_df, test_df, _ = titanic.load(split=True)
 ```
 
-The `ludwig.datasets` API also provides functions to list, describe, and get datasets.  For example:
+The `ludwig.datasets` API provides functions to list, describe, and get datasets:
 
-```python
-import ludwig.datasets
+______________________________________________________________________
+
+### list_datasets
+
+Gets a list of the names of available datasets.
+
+**Example:**
 
-# Gets a list of all available dataset names.
+```python
 dataset_names = ludwig.datasets.list_datasets()
+```
+
+______________________________________________________________________
+
+### describe_dataset
+
+Gets a human-readable description string for a dataset
 
-# Prints the description of the titanic dataset.
+**Example:**
+
+```python
 print(ludwig.datasets.describe_dataset("titanic"))
+```
+
+______________________________________________________________________
+
+### get_dataset
+
+Get a dataset by name
+
+**Example:**
+
+```python
+titanic_dataset = ludwig.datasets.get_dataset("titanic")
+```
+
+______________________________________________________________________
+
+### model_configs_for_dataset
+
+Gets a dictionary of model configs for the specified dataset.  Keys are the config names, and may
+contain the special keys:
+
+- `default` - The default config for the dataset.  Should train to decent performance under 10 minutes on a tyipcal
+  laptop without GPU.
+- `best` - The best known config for the dataset.  Should be replaced when a better config is found.  This is a good
```

Comment:
Could we split this into `best_ecd` and `best_gbm`? A user could be interested in a more lightweight model than a neural network.

Reply:
Lets keep `best` as "the best config we've seen so far" to keep it simple.  Adding GBM configs would be a good follow-up though.